### PR TITLE
Updated README and dependencies

### PR DESCRIPTION
**Pull request by darrenhunt2**
This PR adds the following changes:

- Adds docstrings to files, along with some other minor code cleanup
- Removes the unused `weather.py` file
- Updates the readme
- Updates Node and Python dependencies (needs testing)
- Updates to run with Python 3.9
- Updates the Github Actions (needs testing)

One thing to note is that the dev stage is still called "api" (check the `serverless.yml` file). We should decide if we want to change this so that the stage naming convention matches the other repositories. (Additionally, I added in the readme that the base URL has `/{stage}` at the end, but unsure if this will need to change.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,100 +1,254 @@
...
+Clone the repository to your local machine:
 
-* **Method:**
-  
-  `GET`
-  
-* **Query Params**
+```
+git clone https://github.com/LCOGT/photonranch-api.git
+cd photonranch-calendar
```
Comment (kciurleo) [inline README.md:33]: `cd photonranch-api`

File: README.md [hunk 2]
```diff
@@ -1,100 +1,254 @@
...
-  * **Code:** 500 <br />
-    **Content:** `{ error : "Site config missing required parameters..." }`
+`python -m pytest`
 
-  OR
+## API endpoints
 
-  * **Code:** 500 <br />
-    **Content:** `{ error : "The specified sitecode did not match any config file." }`
+Requests for all services in this repository are handled at the base URL `https://api.photonranch.org/{stage}`.
```
Comment (kciurleo) [inline README.md:64]: I agree that we should change the naming so this is true (and consistent with other repositories) - "dev" for dev, "prod" for prod etc

**Review by kciurleo — APPROVED** [review]
Looks great, especially the endpoint documentation for such varied endpoints!

File: README.md (hunk 1 above)
Comment (darrenhunt2) [reply README.md:33]: Good catch!

File: api/handler.py [hunk 3]
```diff
@@ -200,11 +221,27 @@ def download(event, context):
...
-    """ This method returns a link to download a zip of multiple images in fits format. 
-    First, get a list of files to be zipped based on the query parameters specified. 
-    Next, call a lambda function (defined in the repository zip-downloads) that creates a zip
-    from the list of specified files and uploads that back to s3, returning a presigned download url. 
-    Finally, this function returns the url in the http response to the requester. 
+    """Returns a link to download a zip of multiple images in FITS format.
+
+    First, get a list of files to be zipped based on
+    the query parameters specified. Next, call a Lambda function
+    (defined in the repository zip-downloads) that creates a zip
```
Comment (timbeccue) [inline api/handler.py:228]: The repository is now called photonranch-downloads

File: serverless.yml [hunk 4]
```diff
@@ -144,72 +143,65 @@ resources: # CloudFormation template syntax from here on.
...
           Ref: 'ApiGatewayRestApi'
-        StatusCode: '401'
-
+        StatusCode: '401'    
+        
 functions:
   authorizerFunc: 
-    handler: authorizer.auth
+    handler: api/authorizer.auth
+    cors: true
```
Comment (timbeccue) [inline serverless.yml:151]: this line can be removed

**Review by timbeccue — APPROVED** [review]
Great work! Looks good to me.

File: serverless.yml (hunk 4 above)
Comment (timbeccue) [reply serverless.yml:151]: The `cors: true` line, to be clear

**Pull request merged**